librz/core/cmd/cmd_analysis.c

```diff
@@ -1836,23 +1837,27 @@ RZ_IPI RzCmdStatus rz_analysis_syscall_number_handler(RzCore *core, int argc, co
 }
 
 static void syscall_dump(RzSyscallItem *si, bool is_c) {
+	char tmpbuf[32] = { 0 };
+	snprintf(tmpbuf, sizeof(tmpbuf), SYSCALL_NUM_FMT(si->num), si->num);
 	if (is_c) {
-		rz_cons_printf("#define SYS_%s %s\n", si->name, syscallNumber(si->num));
+		rz_cons_printf("#define SYS_%s %s\n", si->name, tmpbuf);
 	} else {
-		rz_cons_printf(".equ SYS_%s %s\n", si->name, syscallNumber(si->num));
+		rz_cons_printf(".equ SYS_%s %s\n", si->name, tmpbuf);
 	}
 }
 
 static RzCmdStatus syscalls_dump(RzCore *core, int argc, const char **argv, bool is_c) {
+	char tmpbuf[32] = { 0 };
 	if (argc > 1) {
-		st64 n = rz_num_math(core->num, argv[1]);
+		int n = (int)rz_num_math(core->num, argv[1]);
```

Comment:
why? `st64` is signed. but if the issue is the output of `SYSCALL_NUM_FMT` use `PFMT64x` and `PFMT64d`

Reply:
It is used for `int` in some cases, and in `st64` in others, there is no universal solution